packages/tldraw/src/lib/bindings/arrow/ArrowBindingUtil.ts

```diff
@@ -43,12 +43,16 @@ export class ArrowBindingUtil extends BindingUtil<TLArrowBinding> {
 
 	// when the binding itself changes
 	override onAfterCreate({ binding }: BindingOnCreateOptions<TLArrowBinding>): void {
-		arrowDidUpdate(this.editor, this.editor.getShape(binding.fromId) as TLArrowShape)
+		const arrow = this.editor.getShape(binding.fromId) as TLArrowShape | undefined
+		if (!arrow) return
+		arrowDidUpdate(this.editor, arrow)
 	}
 
 	// when the binding itself changes
 	override onAfterChange({ bindingAfter }: BindingOnChangeOptions<TLArrowBinding>): void {
-		arrowDidUpdate(this.editor, this.editor.getShape(bindingAfter.fromId) as TLArrowShape)
+		const arrow = this.editor.getShape(bindingAfter.fromId) as TLArrowShape | undefined
+		if (!arrow) return
```

Comment:
Not sure if we need to do some cleanup here.

Reply:
i don't really understand how we end up with a change event for a binding that i think should be deleted :/ 

we can maybe land this, but i wonder if theres a fix further up the chain? i don't think these bindings should exists

Reply:
Yeah, I also wasn't sure.